Formula/c/cuda@12.9.1.rb

```diff
@@ -0,0 +1,66 @@
+class CudaAT1291 < Formula
+  # CUDA version variables
+  CUDA_VERSION = "12.9.1".freeze
+  CUDA_BUILD = "575.57.08".freeze
+  GCC_VERSION = "14".freeze
+
+  desc "NVIDIA's GPU programming toolkit"
+  homepage "https://developer.nvidia.com/cuda-toolkit"
+
+  on_arm do
+    CUDA_SUFFIX = "_sbsa".freeze
+    CUDA_RUN_SHA256 = "64f47ab791a76b6889702425e0755385f5fa216c5a9f061875c7deed5f08cdb6".freeze
+  end
+  on_intel do
+    CUDA_SUFFIX = "".freeze
+    CUDA_RUN_SHA256 = "0f6d806ddd87230d2adbe8a6006a9d20144fdbda9de2d6acc677daa5d036417a".freeze
+  end
+
+  url "https://developer.download.nvidia.com/compute/cuda/#{CUDA_VERSION}/local_installers/cuda_#{CUDA_VERSION}_#{CUDA_BUILD}_linux#{CUDA_SUFFIX}.run", using: :nounzip
```

Comment:
What is different in the source that there are different downloads for each architecture?

Reply:
It doesn't seem like it's building from source then, which is a requirement for homebrew/core.

Reply:
It does compile, that's why gcc is needed.

Anyway, this provides a compiler (nvcc) which provides GPU acceleration that you can't get with gcc or clang. Not allowing it to be in homebrew-core is really a big limitation because if for example a package wants to compile with cuda, but you don't allow depending on casks in the formula, then the bottles you provide will have limited performance and/or missing features.

Seems like you're trolling me if you are really against having the cuda compiler in what is supposedly a Linux package manager.